src/jdk.jfr/share/classes/jdk/jfr/consumer/RecordedObject.java

```diff
@@ -502,12 +502,11 @@ public final short getShort(String name) {
         if (o instanceof Byte b) {
             return b;
         }
-        if (o instanceof UnsignedValue unsigned) {
```

Comment:
I think this can be further simplified, e.g.,
```java
    return switch(o) {
        case Short s -> s;
        case Byte b -> b;
        case UnsignedValue(Short s) -> s;
        case UnsignedValue(Byte b) -> (short) Byte.toUnsignedInt(b);
        case null, default -> throw newIllegalArgumentException(name, "short");
    };
```
What do you think?

Reply:
I thought about having everything in a switch statements, not just for this method, but I wasn't sure about performance. 99% of the calls will probably be trapped in the first instanceof check. Users are likely to use the method matching the datatype. A fast path might pay off.

I did some measurements and I couldn't spot a difference, so I will change to one  switch statement. Thanks.